wallet/wallet.go

```diff
@@ -1289,6 +1292,14 @@ func WithCustomSelectUtxos(utxos []wire.OutPoint) TxCreateOption {
 	}
 }
 
+// WithAllowUtxo is used to restrict the selection of the internal wallet inputs
+// by further external conditions.
+func WithAllowUtxo(allowUtxo func(utxo wtxmgr.Credit) bool) TxCreateOption {
```

Comment:
suggestion: rename this to `UtxoFilter` as it is fundamentally being used for a Filter operation